cpp/src/table/row_operators.cu

```diff
@@ -264,7 +264,7 @@ auto list_lex_preprocess(table_view table, rmm::cuda_stream_view stream)
   std::vector<detail::dremel_device_view> dremel_device_views;
   for (auto const& col : table) {
     if (col.type().id() == type_id::LIST) {
-      dremel_data.push_back(detail::get_dremel_data(col, {}, false, stream));
+      dremel_data.push_back(detail::get_comparator_data(col, {}, false, stream));
```

Comment:
```suggestion
      dremel_data.emplace_back(detail::get_comparator_data(col, {}, false, stream));
```

Reply:
Also recommend to replace `push_back` by `emplace_back` in many other places, such as the immediate line below.

Reply:
I do not think `emplace_back` will work here. `detail::get_comparator_data` returns `detail::dremel_data&&`. `emplace_back` will instead require `Args&&.. args` which can be used to construct a `detail::dremel_data`

Reply:
You can call `vector<T>::emplace_back(T&&)` which does construct-in-place. Unless your `dremel_data` class is not movable but I don't think so since it is still copied via `push_back`.

Reply:
`push_back` does not create a copy for an r-value reference https://en.cppreference.com/w/cpp/container/vector/push_back